library/Zend/Filter/AbstractDateDropdown.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Filter;
+
+/**
+ * Class AbstractDateDropdown
+ * @package Zend\Filter
+ */
+abstract class AbstractDateDropdown extends AbstractFilter
+{
+    /**
+     * If true, the filter will return null if any date field is empty
+     *
+     * @var bool
+     */
+    protected $nullOnEmpty = false;
+
+    /**
+     * If true, the filter will return null if all date fields are empty
+     *
+     * @var bool
+     */
+    protected $nullOnAllEmpty = false;
+
+    /**
+     * Sprintf format string to use for formatting the date, fields will be used in alphabetical order.
+     *
+     * @var string
+     */
+    protected $format = '';
+
```

Comment:
A constructor should be here, otherwise there is no way setting options to filter. As an example:

```php
/**
 * Constructor
 *
 * @param array|Traversable $options OPTIONAL
 */
public function __construct($options = null)
{
    if (static::isOptions($options)) {
        $this->setOptions($options);
    }
}
```